lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -731,4 +734,81 @@ public void onChannelStateChanged(ChannelStateChange stateChange) {
             assertEquals("Suspended channel histories do not match", suspendedChannelHistory, expectedSuspendedChannelHistory);
         }
     }
+
+    /**
+     * <p>
+     * Verifies that the {@code ConnectionManager} enters the disconnected state and sets the suspend timer
+     * upon unavailable transport.
+     * </p>
+     * <p>
+     * Spec: RTN15g
+     * </p>
+     */
+    @Test
+    public void connection_manager_enters_disconnected_state_on_transport_failure() throws AblyException, NoSuchFieldException, IllegalAccessException {
+        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+        try(AblyRealtime ably = new AblyRealtime(opts)) {
+            ConnectionManager connectionManager = ably.connection.connectionManager;
+            connectionManager.connect();
+            new Helpers.ConnectionManagerWaiter(ably.connection.connectionManager).waitFor(ConnectionState.connected);
+
+            // Here, we "fake" being online for 2 minutes - the suspendTime is set by onConnected and the default is 2 minutes
+            Field suspendTimeField = connectionManager.getClass().getDeclaredField("suspendTime");
+            suspendTimeField.setAccessible(true);
+            suspendTimeField.set(connectionManager, System.currentTimeMillis() - 10);
```

Comment:
Can I ask what's the purpose of this being set to `System.currentTimeMillis() - 10` 🤔